src/main/kotlin/no/nav/tilleggsstonader/sak/vedtak/OpphørValideringService.kt

```diff
@@ -0,0 +1,81 @@
+package no.nav.tilleggsstonader.sak.vedtak
+
+import no.nav.tilleggsstonader.sak.behandling.domain.Saksbehandling
+import no.nav.tilleggsstonader.sak.infrastruktur.exception.brukerfeilHvis
+import no.nav.tilleggsstonader.sak.vedtak.barnetilsyn.domain.BeregningsresultatTilsynBarn
+import no.nav.tilleggsstonader.sak.vilkår.stønadsvilkår.VilkårService
+import no.nav.tilleggsstonader.sak.vilkår.stønadsvilkår.domain.Vilkår
+import no.nav.tilleggsstonader.sak.vilkår.stønadsvilkår.domain.VilkårStatus
+import no.nav.tilleggsstonader.sak.vilkår.stønadsvilkår.domain.Vilkårsresultat
+import no.nav.tilleggsstonader.sak.vilkår.vilkårperiode.VilkårperiodeService
+import no.nav.tilleggsstonader.sak.vilkår.vilkårperiode.domain.ResultatVilkårperiode
+import no.nav.tilleggsstonader.sak.vilkår.vilkårperiode.domain.Vilkårperioder
+import no.nav.tilleggsstonader.sak.vilkår.vilkårperiode.felles.Vilkårstatus
+import org.springframework.stereotype.Service
+import java.time.LocalDate
+
+@Service
+class OpphørValideringService(
+    private val vilkårsperiodeService: VilkårperiodeService,
+    private val vilkårService: VilkårService,
+) {
+
+    fun validerPerioder(saksbehandling: Saksbehandling) {
+        val vilkår = vilkårService.hentVilkår(saksbehandling.id)
+        val vilkårperioder = vilkårsperiodeService.hentVilkårperioder(saksbehandling.id)
+
+        validerIngenNyeOppfylteVilkårEllerVilkårperioder(vilkår, vilkårperioder)
+        validerIngenEndredePerioderMedTomEtterOpphørsdato(
+            vilkårperioder,
+            vilkår,
+            saksbehandling.revurderFra ?: error("Revurder fra er påkrevd for opphør"),
+        )
+    }
+
+    fun validerIngenUtbetalingEtterOpphør(
+        beregningsresultatTilsynBarn: BeregningsresultatTilsynBarn,
+        opphørsDato: LocalDate?,
+    ) {
+        brukerfeilHvis(opphørsDato == null) { "Revurder fra dato er påkrevd for opphør" }
+        beregningsresultatTilsynBarn.perioder.forEach { periode ->
+            periode.beløpsperioder.forEach {
+                brukerfeilHvis(it.dato >= opphørsDato) { "Det er utbetalinger etter opphørsdato" }
```

Comment:
```suggestion
                brukerfeilHvis(it.dato >= opphørsDato) { "Kan ikke velge resultat opphør, fordi det er utbetalinger etter opphørsdato" }
```

Reply:
Her burde vi kanskje gjøre noe mer for å hjelpe saksbehandlerne 🤔. Det er kanskje ikke så lett å vite hvorfor det har blitt en utbetaling

Reply:
`Opphørsdato` er ikke noe vi har i applikasjonen, muligens man burde bruke "revurder fra dato" her og, som på rad 39?

Reply:
Note to self: Kanskje noe om at man prøver å godkjenne nye utbetalinger som skal gå etter opphørsdato ?